addons/base/views.py

```diff
@@ -293,7 +293,7 @@ def get_auth(auth, **kwargs):
         raise HTTPError(httplib.BAD_REQUEST)
 
     node = AbstractNode.load(node_id) or Preprint.load(node_id)
-    if not node:
+    if (node and node.is_deleted) or not node:
```

Comment:
If an AbstractNode or Preprint `is_deleted` (works for both models, one is a property, one is a field), then a 404 is returned at this point in `get_auth`.

This keeps a waterbutler request where the node is deleted from returning a 500 - In this example, if node <h6pjw> is deleted, this will return a 404.
http://localhost:7777/v1/resources/h6pjw/providers/osfstorage/5d62d7d7ac94ea0291356f9f/